src/app/common/context/MustGatherContext.tsx

```diff
@@ -0,0 +1,118 @@
+  name: string;
+  isGathering: boolean;
+}[];
+
+export type mustGatherListType = mustGatherObjType[];
+interface IMustGatherContext {
+  mustGatherModalOpen: boolean;
+  setMustGatherModalOpen: (isOpen: boolean) => void;
+  mustGatherList: mustGatherListType;
+  setActiveMustGather: (mustGatherObj: mustGatherObjType) => void;
+  activeMustGather: mustGatherObjType | undefined;
+  mustGatherWatchList: mgWatchListType;
+  setMustGatherWatchList: (list: mgWatchListType) => void;
+  mustGathers: UseQueryResult<IMustGatherResponse[], Response>;
+}
+
+const mustGatherContextDefaultValue = {} as IMustGatherContext;
+export const MustGatherContext = React.createContext<IMustGatherContext>(
+  mustGatherContextDefaultValue
+);
+
+interface IMustGatherContextProvider {
+  children: React.ReactNode;
+}
+
+export const MustGatherContextProvider: React.FunctionComponent<IMustGatherContextProvider> = ({
+  children,
+}: IMustGatherContextProvider) => {
+  const appContext = React.useContext(AppContext);
+  const [mustGatherModalOpen, setMustGatherModalOpen] = React.useState(false);
+  const [mustGatherList, setMustGatherList] = React.useState<mustGatherListType>([]);
+  const [activeMustGather, setActiveMustGather] = React.useState<mustGatherObjType>();
+  const [mustGatherWatchList, setMustGatherWatchList] = React.useState<mgWatchListType>([]);
+  const [errorCount, setErrorCount] = React.useState(0);
+
+  const mustGathers = useMustGathersQuery(
+    'must-gather',
+    (data) => {
+      const updatedMgList: mustGatherListType = data.map((mg): mustGatherObjType => {
+        return {
+          customName: mg['custom-name'],
+          status: mg.status,
+          type: mg.command.toLowerCase().indexOf('plan') ? 'plan' : 'vm',
+        };
+      });
+      setMustGatherList(updatedMgList);
+      setErrorCount(0);
+    },
+    () => {
+      if (errorCount < 1) {
+        appContext.pushNotification({
+          title: 'Could not reach must gather service.',
+          message: '',
+          key: 'mg-connection-error',
+          variant: 'warning',
+          actionClose: true,
+          timeout: false,
+        });
+      }
+
+      setErrorCount(errorCount + 1);
+    }
+  );
+
+  React.useEffect(() => {
+    const mgToMonitor = mustGatherList.map((mg) => ({
+      name: mg.customName,
+      isGathering: mg.status === 'inprogress',
+    }));
+    setMustGatherWatchList(Array.from(new Set(mgToMonitor)));
+  }, [mustGatherList]);
+
+  return (
+    <MustGatherContext.Provider
+      value={{
+        mustGathers,
+        mustGatherWatchList,
+        setMustGatherWatchList,
+        mustGatherModalOpen,
+        setMustGatherModalOpen,
+        mustGatherList,
+        activeMustGather,
+        setActiveMustGather,
+      }}
+    >
+      <>
+        {children}
+        {mustGatherList.map((mg, idx) => {
+          return (
+            mg.customName && (
+              <div data-watcher={mg.customName} data-status={mg.status} key={idx}>
```

Comment:
I'm on the fence here, but I think we might want to remove this so we don't have unnecessary DOM elements. Considering our other performance issues this is a drop in the bucket though, so I'm fine with leaving it. Either way, I think you can also keep track of watchers by looking at the `must-gather-entity` queries in the react-query dev tools.